src/test/java/org/breedinginsight/brapi/v1/controller/BrapiObservationVariablesControllerIntegrationTest.java

```diff
@@ -0,0 +1,343 @@
+
+        // Insert program
+        dsl.execute(fp.get("InsertProgram"));
+
+        // Insert program observation level
+        dsl.execute(fp.get("InsertProgramObservationLevel"));
+
+        // Insert program ontology sql
+        dsl.execute(fp.get("InsertProgramOntology"));
+        dsl.execute(fp.get("InsertTestProgramUser"));
+        dsl.execute(fp.get("InsertOtherTestProgramUser"));
+
+        // Retrieve our new data
+        validProgram = programDao.findAll().get(0);
+
+        // Insert other program
+        dsl.execute(fp.get("InsertOtherProgram"));
+        dsl.execute(fp.get("InsertOtherProgramObservationLevel"));
+        dsl.execute(fp.get("InsertOtherProgramOntology"));
+        dsl.execute(fp.get("InsertOtherTestOtherProgramUser"));
+
+        otherValidProgram = programDao.fetchByName("Other Test Program").get(0);
+
+        // Add species to BrAPI server
+        fp = FannyPack.fill("src/test/resources/sql/brapi/species.sql");
+        super.getBrapiDsl().execute(fp.get("InsertSpecies"));
+    }
+
+    Trait buildTestTrait(String name, String traitClass) {
+        Method method = Method.builder()
+                .methodName(name + " method")
+                .description(name + " method description")
+                .methodClass("Observation")
+                .build();
+
+        Scale scale = Scale.builder()
+                .scaleName(name + " scale")
+                .dataType(DataType.NUMERICAL)
+                .build();
+
+        ProgramObservationLevel level = ProgramObservationLevel.builder()
+                .name("Plant")
+                .build();
+
+        Trait trait = Trait.builder()
+                .traitName(name)
+                .description(name + " description")
+                .method(method)
+                .scale(scale)
+                .programObservationLevel(level)
+                .traitClass(traitClass)
+                .active(true)
+                .build();
+        return trait;
+    }
+
+    void checkTraits(Trait expected, JsonObject actual) {
+        JsonObject trait = actual.getAsJsonObject("trait");
+        assertEquals(expected.getTraitName(), actual.get("observationVariableName").getAsString(), "Wrong name");
+        assertEquals(expected.getDescription(), trait.get("description").getAsString(), "Wrong description");
+        //assertEquals(validLocation.getCountry().getName(), country.get("name").getAsString(), "Wrong country name");
+    }
+
+    @Test
+    void userDoesNotExist() {
+        Flowable<HttpResponse<String>> call = client.exchange(
+                GET("/variables")
+                        .bearerAuth("non-existent-user"), String.class
+        );
+
+        HttpClientResponseException e = Assertions.assertThrows(HttpClientResponseException.class, () -> {
+            HttpResponse<String> response = call.blockingFirst();
+        });
+        assertEquals(HttpStatus.NOT_FOUND, e.getStatus());
```

Comment:
ya, that would probably be a more descriptive indication of what's going on since it would happen when the token is still valid but the user has been removed from the system.

Reply:
This seems like a bigger problem than just for tokens issued for BrAPI.  Maybe the `SecurityService#getUser` method could be updated to check that the user is in the db and active?

Reply:
I was planning on adding that as part of my program security feature so we can get the roles on each request. I'll see if I can return a 401 if the user isn't found.

Reply:
Sounds good!

Reply:
Changed error code for now and pushed changes, chris will take care of other bit.